src/main/java/net/mcreator/ui/component/JStringListField.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.component;
+
+import net.mcreator.ui.dialogs.ListEditorDialog;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.init.UIRES;
+import net.mcreator.ui.validation.Validator;
+import net.mcreator.ui.validation.component.VTextField;
+
+import javax.annotation.Nullable;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.datatransfer.StringSelection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+public class JStringListField extends JPanel {
+
+	private final List<String> textList = new ArrayList<>();
+	private int index = -1;
+
+	private final JButton edit = new JButton(UIRES.get("16px.edit.gif")) {
+		@Override public String getName() {
+			return "TechnicalButton";
+		}
+	};
+	private final JButton back = new JButton(UIRES.get("previous")) {
+		@Override public String getName() {
+			return "TechnicalButton";
+		}
+	};
+	private final JButton forward = new JButton(UIRES.get("next")) {
+		@Override public String getName() {
+			return "TechnicalButton";
+		}
+	};
+	private boolean joinEntries = false, uniqueEntries = false;

```

Comment:
Ot's only for the style, but I think they should be put before defining buttons, so they are more visible